src/rust/engine/process_execution/src/children.rs

```diff
@@ -20,11 +17,15 @@ const GRACEFUL_SHUTDOWN_POLL_TIME: time::Duration = time::Duration::from_millis(
 /// signals in sequence for https://github.com/pantsbuild/pants/issues/13230.
 pub struct ManagedChild {
   child: Child,
+  graceful_shutdown_max_wait_time: time::Duration,
   killed: AtomicBool,
 }
 
 impl ManagedChild {
-  pub fn spawn(mut command: Command) -> Result<Self, String> {
+  pub fn spawn(
+    mut command: Command,
+    graceful_shutdown_max_wait_time: usize,
```

Comment:
For type safety, should use `Duration` as high in the stack on the rust side as you can: ideally, immediately in `src/rust/engine/src/externs/interface.rs`.